draft-shahzad-scim-device-model.mkd

```diff
@@ -282,6 +282,97 @@ Device groups are created using the SCIM groups as defined in
 {{!RFC7643}} Section 4.2.
 
 
+Resource Type EndpointApp
+=========================
+This section defines a new resource type, 'EndpointApp'. The 
+"ResourceType" schema specifies the metadata about a resource type 
+(see section 6 of {{!RFC7643}}). The resource "EndpointApp" represents
+partner applications that can control and/or receive data from the
+devices. The JSON schema for EndpointApp resource type is in {{resource-schema}}.
+The attributes comprising EndpointsApp are listed in {{endpointapp-schema}}.
+The "EndpointApp" are included in the enpoint applications extension 
+("endpointAppsExt") {{endpointsappext-schema}}.
+
+
+SCIM EndpointApp Schema {#endpointapp-schema}
+=======================
+The schema for "EndpointApp" is identified using the schema URI:
+"urn:ietf:params:scim:schemas:core:2.0:EndpointApp". The following
+attributes are defined in this schema.
+
+## Common Attributes
+
+The EndpointApp schema contains three common attributes as defined in the
+{{!RFC7643}}.
+
+## Singular Attributes
+
+applicationType
+
+This attribute is of type "string" and represents the type of 
+application. It will only contain two values; 'deviceControl' 
+or 'telemetry'. 'deviceControl' is the application send commands
+to control the device. 'telemetry' is the application that receives
+data from the device. The attribute is required, and is not 
+case-sensitive. The is attribute readOnly and should be returned 
+by default. No uniqueness constraints are imposed on this attribute.
+
+applicationName
+
+The "applicationName" attribute is of type "string" and represents a
+human readable name for the application. This attribute is required and
+mutable. The attribute should be returned by default and there is no 
+uniqueness contraint on the attribute.
+
+certificate
+
+This attribute is type "string" and represents a x509certificate that may 
```

Comment:
Do we need to have the certificate of the application in the schema?
We will never use this field for the comparison because we will only check if the client certificate in TLS has a valid signature by a known CA. 

If we need to have any certificate related field then maybe this has to be the CA that was used to sign the certificate? I'm not sure that needs to be in the SCIM resource though.